.github/workflows/bazel-lockfile.yml

```diff
@@ -0,0 +1,221 @@
+          CI_PROJECT_DIR: ${{ github.workspace }}
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+          CI_RUN_ID: ${{ github.run_id }}
+          RUSTFLAGS: "--remap-path-prefix=${CI_PROJECT_DIR}=/ic"
+          BUILDEVENT_DATASET: "github-ci-dfinity"
+        with:
+          BAZEL_COMMAND: "build"
+          BAZEL_TARGETS: "//..."
+          BAZEL_EXTRA_ARGS: '--lockfile_mode=update --nobuild'
+          BAZEL_CI_CONFIG: "--config=ci --repository_cache=/cache/bazel"
+          # check if PR title contains release and set timeout filters accordingly
+          BUILDEVENT_APIKEY: ${{ secrets.HONEYCOMB_API_TOKEN }}
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bazel-module-lock-linux
+          path: ./MODULE.bazel.lock
+
+  # Same a Linux above but with our macOS self-hosted runners.
+  bazel-bzlmod-lockfile-macos-intel:
+    name: Apple Intel
+    needs: bazel-bzlmod-lockfile-linux
+    runs-on:
+      labels: macOS
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set PATH
+        run: |
+          echo "/usr/local/bin" >> $GITHUB_PATH
+          echo "$HOME/.cargo/bin:" >> $GITHUB_PATH
+      - name: Login to Dockerhub
+        shell: bash
+        run: ./ci/scripts/docker-login.sh
+        env:
+          DOCKER_HUB_USER: ${{ vars.DOCKER_HUB_USER }}
+          DOCKER_HUB_PASSWORD_RO: ${{ secrets.DOCKER_HUB_PASSWORD_RO }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: bazel-module-lock-linux
+      - name: Run Bazel Test Darwin x86-64
+        id: bazel-test-darwin-x86-64
+        uses: ./.github/actions/bazel-test-all/
+        env:
+          CI_COMMIT_SHA: ${{ github.sha }}
+          CI_JOB_NAME: ${{ github.job }}
+          CI_JOB_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+          CI_PIPELINE_SOURCE: ${{ github.event_name }}
+          CI_PROJECT_DIR: ${{ github.workspace }}
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+          CI_RUN_ID: ${{ github.run_id }}
+          RUSTFLAGS: "--remap-path-prefix=${CI_PROJECT_DIR}=/ic"
+          BUILDEVENT_DATASET: "github-ci-dfinity"
+        with:
+          BAZEL_CI_CONFIG: "--config=ci --config macos_ci"
+          BAZEL_COMMAND: build
+          BAZEL_EXTRA_ARGS: '--test_tag_filters=test_macos --lockfile_mode=update --nobuild'
+          BAZEL_STARTUP_ARGS: "--output_base /var/tmp/bazel-output/${CI_RUN_ID}"
+          BAZEL_TARGETS: "//rs/... //publish/binaries/..."
+          BUILDEVENT_APIKEY: ${{ secrets.HONEYCOMB_API_TOKEN }}
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bazel-module-lock-apple-intel
+          path: ./MODULE.bazel.lock
+
+  # Finally download the lockfile that went through every platform and check it
+  # for changes. See steps for actual logic depending on GHA event.
+  bazel-bzlmod-lockfile-update:
+    name: Check lockfile changes
```

Comment:
If this is triggered on the merge_group, you will need to add this as a required job as well - otherwise it will be hanging forever.

Reply:
Good point! will do; after merge obviously :)